pkg/gotohelm/gochart.go

```diff
@@ -50,6 +50,25 @@ func Load(chartYAML, defaultValuesYAML []byte, render RenderFunc, dependencies .
 	}, nil
 }
 
+func (c *GoChart) calculateDepsValues() (map[string]any, error) {
+	merged := map[string]any{}
+	for _, dep := range c.dependencies {
+		depsValues, err := dep.calculateDepsValues()
+		if err != nil {
+			return nil, errors.WithStack(errors.Wrapf(err, "failed to calculate dependency value of %q dependency", dep.metadata.Name))
+		}
+
+		depDefaultValues := map[string]any{}
+		err = yaml.Unmarshal(dep.defaultValues, &depDefaultValues)
+		if err != nil {
+			return nil, errors.WithStack(errors.Wrapf(err, "failed to unmarshal %q dependency", dep.metadata.Name))
+		}
+
+		merged[dep.metadata.Name] = helmette.Merge(depDefaultValues, depsValues)
```

Comment:
Any reason to not use `helm.MergeYAMLValues` here? That'd be more consistent with helm itself. Any divergences in `merge` and whatever helm is doing internal could bite us.

Reply:
No reason. I'm used to use `Merge` function to correctly transpile. This package will not be transpiled. I will change that call.

Reply:
Maybe there is only one issue that I need to marshall both input parameters

Reply:
After more thoughts it's fine. I used `helm.MergeYAMLValues`